### PR TITLE
scripts: twisterlib: coverage: exclude `LOG_*` branches

### PR DESCRIPTION
**Pull request by JordanYates**
Disable branch coverage calculations on the `LOG_*` family of macros. Branch misses are due to the implementation of `Z_LOG2` and cannot be reasonably covered in library code. The internal paths through the `Z_LOG2` macro are not of interest when inspecting files that happen to use the Zephyr logging API.

Only implemented for `gcovr`, not sure on the implementation for `lcov`.
This comment suggests it's possible (https://github.com/linux-test-project/lcov/issues/44#issuecomment-427449082)

Previously:
image

With this change:
image

### CODE REVIEW

**Comment by golowanow** [comment]
@JordanYates, quick question: have you considered using exclusion markers instead ?  (GCOVR_EXCL_*)

**Comment by JordanYates** [comment]
> @JordanYates, quick question: have you considered using exclusion markers instead ? (GCOVR_EXCL_*)

I suspect adding `GCOVR_EXCL_BR_LINE` to every location that calls `LOG_*` in the Zephyr tree would not be well received.
Also instead of supporting coverage tool `N+1` being a trivial change, you would need to go through the entire process again.

File: scripts/pylib/twister/twisterlib/coverage.py [hunk 1]
```diff
@@ -278,6 +283,9 @@ def add_ignore_file(self, pattern):
     def add_ignore_directory(self, pattern):
         self.ignores.append(".*/" + pattern + '/.*')
 
+    def add_ignore_branch_pattern(self, pattern):
+        self.ignore_branch_patterns.append(pattern)
+
```
Comment (LukaszMrugala) [inline scripts/pylib/twister/twisterlib/coverage.py:288]: Couldn't this be moved to `CoverageTool` together with `self.ignore_branch_patterns` to make use of inheritance (both `Gcovr` and `Lcov` inherit from it) and reduce code duplication?

Comment (JordanYates) [reply scripts/pylib/twister/twisterlib/coverage.py:288]: It probably could, I just copied the pattern from the other ignore options

Comment (LukaszMrugala) [reply scripts/pylib/twister/twisterlib/coverage.py:288]: Other ignore options actually differ between `Lcov` and `Gcovr` (`*` vs `.*`)

**Review by LukaszMrugala — APPROVED** [review]
I can accept redundant code; it would be nice to change it if another change would be necessary in this PR.

**Review by nashif — APPROVED** [review]

**Pull request merged**